### PR TITLE
Tidy LESS; Add position mixin

### PR DESCRIPTION
**Pull request by owennw**
Part of #191

### CODE REVIEW

File: src/assets/styles/mixins.less [hunk 1]
```diff
@@ -31,30 +31,30 @@
   white-space: nowrap;
 }
 
-.bubble-head(@top: auto, @bottom: auto, @left: auto, @right: auto) {
-  position: absolute;
+.bubble-head(@top: auto, @right: auto, @bottom: auto, @left: auto) {
+  .position(absolute, @top, @right, @bottom, @left);
```
Comment (bjedrzejewski) [inline src/assets/styles/mixins.less]: Is this kind of position mixin not just added boilerplate?

Comment (bjedrzejewski) [inline src/assets/styles/mixins.less]: This is an honest, but bit more general question.

File: src/assets/styles/style.less [hunk 2]
```diff
@@ -32,19 +33,11 @@ body {
 .main {
   background-color: @dark-background;
   padding: 0px;
-  position: fixed;
-  top: 0px;
-  bottom: 0px;
-  left: 0px;
-  right: 0px;
+  .position(fixed, 0px, 0px, 0px, 0px);
```
Comment (bjedrzejewski) [inline src/assets/styles/style.less]: Without named parameters this is less readable than before.

File: src/assets/styles/style.less [hunk 3]
```diff
@@ -32,19 +33,11 @@ body {
...
 }
 
 .main-content {
-  bottom: -12px;
-  position: absolute;
-  top: @toolbar-height;
-  left: (@sidebar-width + 20);
-  right: -2px;
+  .position(absolute, @toolbar-height, -2px, -12px, (@sidebar-width + 20));
```
Comment (bjedrzejewski) [inline src/assets/styles/style.less]: Lack of names parameters is making this mysterious :)

File: src/assets/styles/style.less [hunk 4]
```diff
@@ -61,10 +54,8 @@ body {
 
 #showcase-title {
   color: @chart-teal;
-  position: absolute;
-  right: 75px;
+  .position(absolute, @top: -10px, @right: 75px);
```
Comment (bjedrzejewski) [inline src/assets/styles/style.less]: Named parameters...

File: src/assets/styles/style.less (hunk 2 above)
Comment (owennw) [inline src/assets/styles/style.less]: I can add the names in. I guess I was leveraging the idea of 'top, right, bottom, left' as is common with padding, margins etc.

File: src/assets/styles/mixins.less (hunk 1 above)
Comment (owennw) [inline src/assets/styles/mixins.less]: Not sure if I understand -- it's like any other helper function. There are 4 lines below that are removed with the use of the new mixin here.

File: src/showcase/showcase-changes.less [hunk 5]
```diff
@@ -99,11 +98,11 @@
   }
 
   .y-axis {
-      .tick {
-          &.orient-right text {
-            .font(10px);
-          }
+    .tick {
+      &.orient-right text {
+        .font(10px);
```
Comment (bjedrzejewski) [inline src/showcase/showcase-changes.less]: Is making a mixin for font sizes and styles a good idea?

File: src/sidebars/favourites/minichart/minichart.less [hunk 6]
```diff
@@ -38,8 +39,8 @@
 }
 
 .minichart-error {
-    padding: 0px 3px;
-    text-align: center;
-    .font(12px, @font-thin);
-    color: white;
+  padding: 0px 3px;
+  text-align: center;
+  .font(12px, @font-thin);
```
Comment (bjedrzejewski) [inline src/sidebars/favourites/minichart/minichart.less]: Mixin for fonts?

File: src/showcase/showcase-changes.less (hunk 5 above)
Comment (owennw) [inline src/showcase/showcase-changes.less]: Why wouldn't it be? In any case I didn't change that, just updated the indentation.

File: src/sidebars/favourites/minichart/minichart.less [hunk 7]
```diff
@@ -38,8 +39,8 @@
...
+  color: white;
```
Comment (bjedrzejewski) [inline src/sidebars/favourites/minichart/minichart.less]: Maybe we can make it a variable for a better less structure?

File: src/sidebars/search/search.less [hunk 8]
```diff
@@ -46,7 +46,6 @@
     width: @expanded-width;
     border-bottom: @search-splitter-style;
 
-
     .name {
       .font(12px, @font-thin);
```
Comment (bjedrzejewski) [inline src/sidebars/search/search.less]: Font mixin maybe?

File: src/sidebars/favourites/minichart/minichart.less (hunk 6 above)
Comment (owennw) [inline src/sidebars/favourites/minichart/minichart.less]: I don't understand the comment. That is a mixin already. (Again, not trying to pass the buck here, but all I did was update the whitespace!)

File: src/assets/styles/style.less (hunk 2 above)
Comment (owennw) [inline src/assets/styles/style.less]: Thinking about it we should add the names. It's more readable and more resilient to the function changing in the future.

File: src/sidebars/favourites/minichart/minichart.less (hunk 7 above)
Comment (bjedrzejewski) [inline src/sidebars/favourites/minichart/minichart.less]: I mean just the color

File: src/showcase/showcase-changes.less (hunk 5 above)
Comment (bjedrzejewski) [inline src/showcase/showcase-changes.less]: Yeah, sorry I was confused. Consider this closed.

File: src/sidebars/star/star.less [hunk 9]
```diff
@@ -8,32 +8,30 @@
...
   z-index: 1050;
-  display: none;
   .no-overflow;
   outline: 0;
   display: block;
+
   .confirmation-modal {
-    &.flipped{
+    &.flipped {
       .box-shadow(-2px -2px, @dark-background);
```
Comment (bjedrzejewski) [inline src/sidebars/star/star.less]: We use this shadow twice, maybe this is a variable rather than mixin? Or a variable that is a mixin!

**Comment by bjedrzejewski** [comment]
@owennw couple comments, other than that, looks great.

File: src/sidebars/star/star.less (hunk 9 above)
Comment (owennw) [inline src/sidebars/star/star.less]: It's currently used 3 times. Once for the search tab (2px 0px - right), once on the top confirmation box (-2px -2px - top left) and on the flipped confirmation box (-2px 2px - bottom left).

These 3 could be abstracted into constants perhaps? Not sure if it's worth it though.

Comment (bjedrzejewski) [inline src/sidebars/star/star.less]: Not worth it. I guess its more the number '2px' being kind of a part of our shadow `design unit` different issue though.

**Pull request merged**